src/satosa/base.py

```diff
@@ -276,6 +277,15 @@ def expose_entityid_endpoint(self):
         value = self.config.get(self.KEY_ENTITYID_ENDPOINT, False)
         return bool(value)
 
+    def enable_metadata_reload(self):
+        """
+        Check whether metadata reload has been enabled in config
+
+        return: bool
+        """
+        value = self.config.get(self.KEY_ENABLE_METADATA_RELOAD, False)
+        return bool(value)
+
```

Comment:
Add generic form

```suggestion

    def _reload_metadata(self, context, config):
        """
        Reload SAML metadata
        """
        logger.debug("Reloading metadata")
        res = self.idp.reload_metadata(copy.deepcopy(config))
        message = "Metadata reload %s" % ("OK" if res else "failed")
        status = "200 OK" if res else "500 FAILED"
        return Response(message=message, status=status)
```

Reply:
The generic form has a different signature

```
    def _reload_metadata(self, context, config):
```

vs

```
    def _reload_metadata(self, context):
```

maybe `SAMLBackend._reload_metadata` and `SAMLFrontend._reload_metadata` should have a different name.

On the suggestions, I call `super()._reload_metadata` to differentiate the two. If they are renamed, we can call `self._reload_metadata(context, config)` directly.